src/Tvheadend.cpp

```diff
@@ -2158,6 +2165,23 @@ bool CTvheadend::ParseEvent ( htsmsg_t *msg, bool bAdd, Event &evt )
   /* Recordings complete */
   SyncDvrCompleted();
 
+  if (htsmsg_get_s64(msg, "stop", &stop) && bAdd)
+  {
+    Logger::Log(LogLevel::LEVEL_ERROR, "malformed eventAdd: 'stop' missing");
+    return false;
+  }
+
+  // tvheadend's "epgMaxTime" implementation is buggy in htsp versions prior to 25.
+  // To workaround this, we must request all epg data and filter it here.
+  if (m_conn.GetProtocol() < 25 &&
+      Settings::GetInstance().GetAsyncEpg() &&
+      m_epgMaxDays > EPG_TIMEFRAME_UNLIMITED)
+  {
+    // check event's stop time against m_epgMaxDays, return if to large
+    if (stop > static_cast<int64_t>(time(NULL) + m_epgMaxDays * int64_t(24 * 60 *60)))
+      return false;
+  }
+
```

Comment:
I would leave it as it is for now, Especially for Krypton. For v18 we can remove the setting entirely to avoid confusion and incorrect behaviour when enabling the setting for older backends.